src/pyclass.rs

```diff
@@ -276,15 +252,17 @@ where
     }
 }
 
-/// Register `T: PyClass` to Python interpreter.
 #[cfg(not(Py_LIMITED_API))]
-pub fn initialize_type<T>(py: Python, module_name: Option<&str>) -> PyResult<*mut ffi::PyTypeObject>
+pub fn create_type_object<T>(
+    py: Python,
+    module_name: Option<&str>,
+) -> PyResult<Box<ffi::PyTypeObject>>
 where
     T: PyClass,
 {
-    let type_object: &mut ffi::PyTypeObject = unsafe { T::type_object() };
-    let base_type_object: &mut ffi::PyTypeObject =
-        unsafe { <T::BaseType as PyTypeInfo>::type_object() };
+    let mut boxed = Box::new(ffi::PyTypeObject_INIT);
```

Comment:
Let's comment why we need `Box` here(i.e., because `PyType_Ready` reuires a constant address).